packages/nuxt/src/app/composables/router.ts

```diff
@@ -56,22 +56,56 @@ export interface NavigateToOptions {
   redirectCode?: number
 }
 
+const getPath = (to: RouteLocationRaw): string => {
+  if (typeof to === 'string') {
+    return to
+  }
+  if ('path' in to) {
+    return to.path
+  }
+  return ''
+}
+
+const isExternalLink = (str: string): boolean => {
+  try {
+    // eslint-disable-next-line no-new
+    new URL(str)
+    return true
+  } catch {
+    return false
+  }
+}
+
 export const navigateTo = (to: RouteLocationRaw, options: NavigateToOptions = {}): Promise<void | NavigationFailure> | RouteLocationRaw => {
   if (!to) {
     to = '/'
   }
+
+  const path = getPath(to)
+  const isExternal = isExternalLink(path)
+
   if (isProcessingMiddleware()) {
```

Comment:
I don't believe vue-router supports redirection to an external URL, so we would need to handle this in the same way as below...

```suggestion
  if (!isExternal && isProcessingMiddleware()) {
```

Reply:
Sounds good! Added it to the PR 👍